### ISSUE TITLE
MetadataCommand.ExtractMetadata fails for C# project if it has a C# project reference on macOS

### ISSUE THREAD

**Issue by bentayloruk** [dotnet/docfx#3746]
**Updated:  I thought this issue was limited to when a C# project had a reference to an F# project, but in fact it occurs when the project reference is C# or F#.**

**DocFX Version Used**:  2.40.4 (from brew).

**Mono Version**: Mono JIT compiler version 5.16.0.220

**macOS**:  Mojave 10.14.1

**Template used**: `default`

**Steps to Reproduce**:

1.  Download this [minimal repro C# project with C# project reference](https://github.com/dotnet/docfx/files/2695053/DocFXFailsWithProjRef.zip) or the original [minimal repro of a C# project with F# project reference](https://github.com/dotnet/docfx/files/2658011/DocFXFSharpProjRef.zip).
2.  Unzip repro solution and run `docfx docfx_project/docfx.json` in the root dir.

Or...

1. Create a new solution.
2. Add a C# class library targetting `netstandard2.0`.
3. Add an F# class library targetting `netstandard2.0`.
4.  Reference the F# class library from the C# class library, using a project reference.
4. Setup a docfx project to with a metadata command targetting the csproj.
5. Run docfx and you will see an error.
6.  If you change the reference from a project reference to an assembly reference, it works fine.

**Expected Behavior**:  I would expect docfx to be able to generate documentation for my C# project without this error.

**Actual Behavior**:  There is an error when a C# project has a project reference to an F# project.

```
15:46 $ docfx docfx_project/docfx.json
[18-12-07 03:47:12.334]Info:[MetadataCommand.ExtractMetadata]Loading projects...
[18-12-07 03:47:14.830]Warning:[MetadataCommand.ExtractMetadata](/Users/mrt/@build/DocFXFSharpProjRef/MyDocProject/MyDocProject.csproj)Workspace failed with: [Failure] Msbuild failed when processing the file '/Users/mrt/@build/DocFXFSharpProjRef/MyDocProject/MyDocProject.csproj' with message: /Library/Frameworks/Mono.framework/Versions/5.16.0/lib/mono/msbuild/15.0/bin/Microsoft.Common.CurrentVersion.targets: (1677, 5): The "GetReferenceNearestTargetFrameworkTask" task failed unexpectedly.
System.TypeLoadException: Could not resolve type with token 01000026 from typeref (expected class 'NuGet.Common.INuGetLogMessage' in assembly 'NuGet.Common, Version=4.8.0.1, Culture=neutral, PublicKeyToken=31bf3856ad364e35')
  at NuGet.Common.LoggerBase.Log (NuGet.Common.LogLevel level, System.String data) [0x00011] in <44b4648adea14c1d8db57b20945560a8>:0 
  at NuGet.Build.Tasks.BuildTasksUtility.LogTaskParam (NuGet.Common.ILogger log, System.String direction, System.String name, System.String[] values) [0x0004e] in <8b71f76b40134dbb897d969ae864ca47>:0 
  at NuGet.Build.Tasks.BuildTasksUtility.LogInputParam (NuGet.Common.ILogger log, System.String name, System.String[] values) [0x00000] in <8b71f76b40134dbb897d969ae864ca47>:0 
  at NuGet.Build.Tasks.GetReferenceNearestTargetFrameworkTask.Execute () [0x00020] in <8b71f76b40134dbb897d969ae864ca47>:0 
  at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute () [0x00029] in <6044aaf8735b4366988ad32930ae4233>:0 
  at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask (Microsoft.Build.BackEnd.ITaskExecutionHost taskExecutionHost, Microsoft.Build.BackEnd.Logging.TaskLoggingContext taskLoggingContext, Microsoft.Build.BackEnd.TaskHost taskHost, Microsoft.Build.BackEnd.ItemBucket bucket, Microsoft.Build.BackEnd.TaskExecutionMode howToExecuteTask) [0x00239] in <6044aaf8735b4366988ad32930ae4233>:0 
[18-12-07 03:47:15.135]Warning:[MetadataCommand.ExtractMetadata](/Users/mrt/@build/DocFXFSharpProjRef/MyDocProject/MyDocProject.csproj)Workspace failed with: [Failure] Cannot open project '/Users/mrt/@build/DocFXFSharpProjRef/FSharpRefProj/FSharpRefProj.fsproj' because the file extension '.fsproj' is not associated with a language.
[18-12-07 03:47:15.261]Info:[MetadataCommand.ExtractMetadata]Generating metadata for each project...
[18-12-07 03:47:15.746]Warning:[MetadataCommand.ExtractMetadata]No metadata is generated for MyDocProject.
[18-12-07 03:47:15.748]Info:[MetadataCommand]Completed Scope:MetadataCommand in 3763.6872 milliseconds.
[18-12-07 03:47:15.928]Info:[BuildCommand]6 plug-in(s) loaded.
[18-12-07 03:47:15.959]Info:[BuildCommand]No files are found with glob pattern apidoc/**.md, excluding obj/**,_site/**, under directory "/Users/mrt/@build/DocFXFSharpProjRef/docfx_project"
[18-12-07 03:47:15.961]Info:[BuildCommand]No files are found with glob pattern images/**, excluding <none>, under directory "/Users/mrt/@build/DocFXFSharpProjRef/docfx_project"
[18-12-07 03:47:15.980]Info:[BuildCommand]Markdown engine is markdig
[18-12-07 03:47:16.166]Info:[BuildCommand]Cannot load build info: 'build.info' not found under '/Users/mrt/@build/DocFXFSharpProjRef/docfx_project/obj/.cache/build'
[18-12-07 03:47:16.181]Info:[BuildCommand.BuildCore.Build Document]Max parallelism is 8.
[18-12-07 03:47:16.228]Info:[BuildCommand.BuildCore.Build Document.Prepare.CreateIncrementalBuildContext]Build strategy: IsFullBuild
[18-12-07 03:47:16.434]Info:[BuildCommand.BuildCore.Build Document.CompilePhaseHandlerWithIncremental.TocDocumentProcessor]Building 3 file(s) in TocDocumentProcessor(BuildTocDocument)...
[18-12-07 03:47:16.434]Info:[BuildCommand.BuildCore.Build Document.CompilePhaseHandlerWithIncremental.ConceptualDocumentProcessor]Building 3 file(s) in ConceptualDocumentProcessor(BuildConceptualDocument=>CountWord=>ValidateConceptualDocumentMetadata)...
[18-12-07 03:47:17.093]Info:[BuildCommand.BuildCore.Build Document.LinkPhaseHandlerWithIncremental.Apply Templates]Applying templates to 6 model(s)...
[18-12-07 03:47:17.229]Info:[BuildCommand.BuildCore.Build Document]XRef map exported.
[18-12-07 03:47:17.310]Info:[BuildCommand.Postprocess]Manifest file saved to manifest.json.
[18-12-07 03:47:17.374]Info:[BuildCommand]Completed building documents in 1434.6621 milliseconds.
[18-12-07 03:47:17.375]Info:[BuildCommand]Completed Scope:BuildCommand in 1626.5381 milliseconds.
[18-12-07 03:47:17.375]Info:Completed in 5393.6203 milliseconds


Build succeeded with warning.
[18-12-07 03:47:17.378]Warning:[MetadataCommand.ExtractMetadata](/Users/mrt/@build/DocFXFSharpProjRef/MyDocProject/MyDocProject.csproj)Workspace failed with: [Failure] Msbuild failed when processing the file '/Users/mrt/@build/DocFXFSharpProjRef/MyDocProject/MyDocProject.csproj' with message: /Library/Frameworks/Mono.framework/Versions/5.16.0/lib/mono/msbuild/15.0/bin/Microsoft.Common.CurrentVersion.targets: (1677, 5): The "GetReferenceNearestTargetFrameworkTask" task failed unexpectedly.
System.TypeLoadException: Could not resolve type with token 01000026 from typeref (expected class 'NuGet.Common.INuGetLogMessage' in assembly 'NuGet.Common, Version=4.8.0.1, Culture=neutral, PublicKeyToken=31bf3856ad364e35')
  at NuGet.Common.LoggerBase.Log (NuGet.Common.LogLevel level, System.String data) [0x00011] in <44b4648adea14c1d8db57b20945560a8>:0 
  at NuGet.Build.Tasks.BuildTasksUtility.LogTaskParam (NuGet.Common.ILogger log, System.String direction, System.String name, System.String[] values) [0x0004e] in <8b71f76b40134dbb897d969ae864ca47>:0 
  at NuGet.Build.Tasks.BuildTasksUtility.LogInputParam (NuGet.Common.ILogger log, System.String name, System.String[] values) [0x00000] in <8b71f76b40134dbb897d969ae864ca47>:0 
  at NuGet.Build.Tasks.GetReferenceNearestTargetFrameworkTask.Execute () [0x00020] in <8b71f76b40134dbb897d969ae864ca47>:0 
  at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute () [0x00029] in <6044aaf8735b4366988ad32930ae4233>:0 
  at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask (Microsoft.Build.BackEnd.ITaskExecutionHost taskExecutionHost, Microsoft.Build.BackEnd.Logging.TaskLoggingContext taskLoggingContext, Microsoft.Build.BackEnd.TaskHost taskHost, Microsoft.Build.BackEnd.ItemBucket bucket, Microsoft.Build.BackEnd.TaskExecutionMode howToExecuteTask) [0x00239] in <6044aaf8735b4366988ad32930ae4233>:0 
[18-12-07 03:47:17.379]Warning:[MetadataCommand.ExtractMetadata](/Users/mrt/@build/DocFXFSharpProjRef/MyDocProject/MyDocProject.csproj)Workspace failed with: [Failure] Cannot open project '/Users/mrt/@build/DocFXFSharpProjRef/FSharpRefProj/FSharpRefProj.fsproj' because the file extension '.fsproj' is not associated with a language.
[18-12-07 03:47:17.379]Warning:[MetadataCommand.ExtractMetadata]No metadata is generated for MyDocProject.
	3 Warning(s)
	0 Error(s)


```

**Comment by bentayloruk** [comment]
I've updated this issue as it turns out it's not limited to F# project references.  A simple C# project that references another C# project is failing in the same way, so maybe it's macOS mono related.  

I've tried looking in to this, but I'm not that familiar with MSBuild.  I did notice [this issue](https://github.com/Microsoft/msbuild/issues/2753) and wondered if it was in some way related, but only because they mention `GetReferenceNearestTargetFrameworkTask` and Nuget, which both crop up in the error I'm seeing here (and because in the msbuild issue, they mention this task failing in non VS environments).
